### PR TITLE
moving average: new moving average EMA/WMA/HMA

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Remove unnecessary data strust and add new moving average.

It refers to #1955

It is the third steps of #3180

It can be reviewed after ##3181 merged.

### Check List

Tests

- Unit test

### CODE REVIEW

**Review by HunDunDM — APPROVED** [review]

**Review by disksing — APPROVED** [review]

**Comment by hi-rustin** [comment]
test-integration-lightning-test

**Comment by hi-rustin** [comment]
> /rebase

No need to rebase and rebase actions already removed.

**Review by Yisaer — COMMENTED** [review]
/merge

File: pkg/movingaverage/moving_average_test.go [hunk 1]
```diff
@@ -81,3 +82,33 @@ func (t *testMovingAvg) TestMedianFilter(c *C) {
...
 }
+
+type testCase struct {
+	ma       MovingAvg
+	expected []float64
+}
+
+func (t *testMovingAvg) TestMovingAvg(c *C) {
+	var empty float64 = 0
+	data := []float64{120, 130, 140, 150, 145, 136, 121, 132, 145, 156, 148, 157, 175}
```
Comment (rleungx) [inline pkg/movingaverage/moving_average_test.go:93]: This code may be hard to maintain in the later just like the tests of the hot region

**Pull request merged**